pkg/lightning/backend/kv/context.go

```diff
@@ -0,0 +1,260 @@
+	"github.com/pingcap/tidb/pkg/util/intest"
+	"github.com/pingcap/tidb/pkg/util/timeutil"
+)
+
+var _ exprctx.ExprContext = &litExprContext{}
+
+// litExprContext implements the `exprctx.ExprContext` interface for lightning import.
+// It provides the context to build and evaluate expressions, furthermore, it allows to set user variables
+// for `IMPORT INTO ...` statements.
+type litExprContext struct {
+	*contextstatic.StaticExprContext
+	userVars *variable.UserVars
+}
+
+// NewExpressionContext creates a new `*StaticExprContext` for lightning import.
+func newLitExprContext(sqlMode mysql.SQLMode, sysVars map[string]string, timestamp int64) (*litExprContext, error) {
+	flags := types.DefaultStmtFlags.
+		WithTruncateAsWarning(!sqlMode.HasStrictMode()).
+		WithIgnoreInvalidDateErr(sqlMode.HasAllowInvalidDatesMode()).
+		WithIgnoreZeroInDate(!sqlMode.HasStrictMode() || sqlMode.HasAllowInvalidDatesMode() ||
+			!sqlMode.HasNoZeroInDateMode() || !sqlMode.HasNoZeroDateMode())
+
+	errLevels := stmtctx.DefaultStmtErrLevels
+	errLevels[errctx.ErrGroupTruncate] = errctx.ResolveErrLevel(flags.IgnoreTruncateErr(), flags.TruncateAsWarning())
+	errLevels[errctx.ErrGroupBadNull] = errctx.ResolveErrLevel(false, !sqlMode.HasStrictMode())
+	errLevels[errctx.ErrGroupDividedByZero] =
+		errctx.ResolveErrLevel(!sqlMode.HasErrorForDivisionByZeroMode(), !sqlMode.HasStrictMode())
+
+	userVars := variable.NewUserVars()
+	evalCtx := contextstatic.NewStaticEvalContext(
+		contextstatic.WithSQLMode(sqlMode),
+		contextstatic.WithTypeFlags(flags),
+		contextstatic.WithLocation(timeutil.SystemLocation()),
+		contextstatic.WithErrLevelMap(errLevels),
+		contextstatic.WithUserVarsReader(userVars),
+	)
+
+	// no need to build as plan cache.
+	planCacheTracker := contextutil.NewPlanCacheTracker(contextutil.IgnoreWarn)
+	intest.Assert(!planCacheTracker.UseCache())
+	ctx := contextstatic.NewStaticExprContext(
+		contextstatic.WithEvalCtx(evalCtx),
+		contextstatic.WithPlanCacheTracker(&planCacheTracker),
+	)
+
+	if len(sysVars) > 0 {
+		var err error
+		ctx, err = ctx.LoadSystemVars(sysVars)
+		if err != nil {
+			return nil, err
+		}
+		evalCtx = ctx.GetStaticEvalCtx()
+	}
+
+	currentTime := func() (time.Time, error) { return time.Now(), nil }
+	if timestamp > 0 {
+		currentTime = func() (time.Time, error) { return time.Unix(timestamp, 0), nil }
+	}
+
+	evalCtx = evalCtx.Apply(contextstatic.WithCurrentTime(currentTime))
+	ctx = ctx.Apply(contextstatic.WithEvalCtx(evalCtx))
+
+	return &litExprContext{
+		StaticExprContext: ctx,
+		userVars:          userVars,
+	}, nil
+}
+
+// setUserVarVal sets the value of a user variable.
+func (ctx *litExprContext) setUserVarVal(name string, dt types.Datum) {
+	ctx.userVars.SetUserVarVal(name, dt)
+}
+
+// UnsetUserVar unsets a user variable.
+func (ctx *litExprContext) unsetUserVar(varName string) {
+	ctx.userVars.UnsetUserVar(varName)
+}
+
+var _ table.MutateContext = &litTableMutateContext{}
```

Comment:
not related to this PR, I prefer put MutateContext into a pkg with more semantic meaning, a pkg of name `context` is so common, it's hard to infer what's inside

a example would be: put it into pkg named as `mutate`, and its name is `Context`, then we can use it as `mutate.Context`, no need for type alias

Reply:
ok, I'll do it in other PRs